src/Analyzers/Core/Analyzers/AddAccessibilityModifiers/AbstractAddAccessibilityModifiersDiagnosticAnalyzer.cs

```diff
@@ -17,6 +18,18 @@ internal abstract class AbstractAddAccessibilityModifiersDiagnosticAnalyzer<TCom
         new LocalizableResourceString(nameof(AnalyzersResources.Accessibility_modifiers_required), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)))
     where TCompilationUnitSyntax : SyntaxNode
 {
+    protected abstract IAccessibilityFacts AccessibilityFacts { get; }
+    protected abstract IAddAccessibilityModifiers AddAccessibilityModifiers { get; }
+
+    protected abstract void ProcessCompilationUnit(SyntaxTreeAnalysisContext context, CodeStyleOption2<AccessibilityModifiersRequired> option, TCompilationUnitSyntax compilationUnitSyntax);
+
+    protected readonly DiagnosticDescriptor ModifierRemovedDescriptor = CreateDescriptorWithId(
+        IDEDiagnosticIds.AddAccessibilityModifiersDiagnosticId,
+        EnforceOnBuildValues.AddAccessibilityModifiers,
+        hasAnyCodeStyleOption: true,
+        new LocalizableResourceString(nameof(AnalyzersResources.Remove_accessibility_modifiers), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
+        new LocalizableResourceString(nameof(AnalyzersResources.Accessibility_modifiers_unnecessary), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)));
```

Comment:
@sharwell is this the correct way to have the analyzer have distinct diagnostic messages?

Reply:
Essentially yes

Reply:
so is this PR good to go then? @sharwell ? :)